v2/extensions/dataref_extension_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package extensions
+
+import (
+	"testing"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+)
+
+func TestAddDataRefExtension(t *testing.T) {
+	// MUST HAVE AT LEAST ONE FAILING TEST as that'll also test to make sure
+	// that the failed add() call won't actually set anything
+
+	tests := []struct {
+		dataref string
+		pass    bool
+	}{
+		{"https://example.com/data", true},
+		{"://invalid-url", false},
+	}
+
+	for _, test := range tests {
+		e := event.New()
+		err := AddDataRefExtension(&e, test.dataref)
+
+		// Make sure adding it passed/fails appropriately
+		if test.pass && err != nil {
+			t.Fatalf("Failed to add DataRefExtension with valid URL(%s): %s",
+				test.dataref, err)
+		}
+		if !test.pass && err == nil {
+			t.Fatalf("Expected not to find DataRefExtension (%s), but did",
+				test.dataref)
+		}
+
+		// Now make sure it's actually there in the 'pass' cases, but
+		// missing in the failed cases.
+		dr, ok := GetDataRefExtension(e)
+		if test.pass {
+			if !ok {
+				t.Fatalf("Dataref (%s) is missing after being set",
+					test.dataref)
+			}
+			if dr == nil || dr.DataRef != test.dataref {
+				t.Fatalf("Retrieved dataref(%v) doesn't match set value(%s)",
+					dr, test.dataref)
+			}
+		} else {
```

Comment:
nit: I think it's safe to remove the `else` and directly jump to the next `if` without changing test semantics?

Reply:
I think we need it because we only want to fail when the dataref is set on the "failed" case, not when the pass is expected to pass.